scripts/get_target_distance_service.py

```diff
@@ -0,0 +1,116 @@
+    - target_position: Robot's target set by the user as RobotTarget message.
+    - isTargetSet: Flag indicating whether the target has been set, for the first time
+    
+    Subscribers:
+    - "/robot_target": Subscribes to the robot's target position updates.
+    - "/robot_state": Subscribes to the robot's current state updates.
+
+    Service:
+    - "get_target_distance": Provides distance and average speed information based on the robot's state and target.
+
+    Methods:
+    - robot_target_callback: Callback for the "/robot_target" subscriber, stores the latest robot's target.
+    - robot_state_callback: Callback for the "/robot_state" subscriber, collects current position and values for average speed.
+    - handle_get_target_distance: Handles the "get_target_distance" service, calculates distance and average speed based on the robot's state and target.
+    """
+
+    def __init__(self, name: str) -> None:
+        rospy.init_node(name, anonymous=False)  
+        
+        self.robot_speed_list = []                                                     
+        self.av_speed_window_limit = rospy.get_param('av_window_size', default=10)      
+        self.robot_current = RobotState()                                               
+        self.target_position = RobotTarget()                                            
+        self.isTargetSet = False      
+
+        rospy.Subscriber("/robot_target", RobotTarget, self.robot_target_callback)     
+        rospy.Subscriber("/robot_state", RobotState, self.robot_state_callback)
+        self.service = rospy.Service("get_target_distance", TargetDistance, \
+                                      self.handle_get_target_distance)
+                                      
+
+    def robot_target_callback(self, data):
+        # store the latest robot's target
+        self.target_position.x = data.x
+        self.target_position.y = data.y
+        # When first target arrives, set to True
+        if not self.isTargetSet:
+
+            self.isTargetSet = True
+
+    def robot_state_callback(self, data):
+        """Callback to robot_state subscriber, collects current position and values for av_speed"""
+        self.robot_current.x_pos = data.x_pos
+        self.robot_current.y_pos = data.y_pos
+        # av_speed window
+        if len(self.robot_speed_list) < self.av_speed_window_limit:
+            self.robot_speed_list.append((data.x_vel, data.y_vel))
+        elif len(self.robot_speed_list) == self.av_speed_window_limit:
+            # First remove the first element
+            self.robot_speed_list.pop(0)
+            # Add data at the end of the list
+            self.robot_speed_list.append((data.x_vel, data.y_vel))
+        else:
+            rospy.logerr("The list size is goes over the theshold.")
+            rospy.logerr("List size = %d, Threshold is = %d", len(self.robot_speed_list), self.av_speed_window_limit)
```

Comment:
```suggestion
            rospy.logerr(f"List size = {len(self.robot_speed_list)}, Threshold is = {self.av_speed_window_limit}")
```
If you are using python 3 for gods sake use this syntax